DiagnosticsExtension/Models/ConnectionStringValidator/IConnectionStringValidator.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DiagnosticsExtension.Models.ConnectionStringValidator
+{
+    interface IConnectionStringValidator
+    {
+        // verify provided string is a valid connection string that can be tested by the validator
+        bool IsValid(string connStr);
+
+        Task<ConnectionStringValidationResult> Test(string connStr, string clientId = null);  // clientId used for Used Assigned Managed Identity
```

Comment:
nit: name this function "Validate" or "ValidateConnectionString" to align with the interface name

Reply:
fixed